Components/UserCamera.js

```diff
@@ -0,0 +1,72 @@
+import React, { useState } from 'react'
+import styled from 'styled-components/native'
+import { TouchableOpacity, Text, View } from 'react-native'
+import { Camera } from 'expo-camera'
+export const UserCamera = (props) => {
+    const { photos, setPhotos, setShowCamera } = props
+    const [type, setType] = useState(Camera.Constants.Type.back);
+    const [cameraRef, setCameraRef] = useState(null);
+
+    const CameraButtons = styled.View`
+flex-direction: row;
+justify-content: space-around;
+align-items: center;
+margin-bottom: 20px;
+`
+
+    return (
+        <Camera style={{ flex: 1, justifyContent: 'flex-end' }} type={type} ref={ref => {
+            setCameraRef(ref);
+        }}>
+            <CameraButtons>
+                <TouchableOpacity
+                    onPress={() => {
+                        setType(
+                            type === Camera.Constants.Type.back
+                                ? Camera.Constants.Type.front
+                                : Camera.Constants.Type.back
+                        );
+                    }}>
+                    <Text style={{ fontSize: 30, color: 'white' }}> Flip </Text>
+                </TouchableOpacity>
+                <TouchableOpacity
+                    style={{ alignSelf: 'flex-end' }}
+                    onPress={async () => {
+                        if (cameraRef) {
+                            let photo = await cameraRef.takePictureAsync();
+                            if (!photo.cancelled && photos.length < 14) {
+                                setPhotos(photos => [photo.uri, ...photos])
+                                setPhotos(photos => [photo.uri, ...photos])
```

Comment:
what are you doing here calling this setPhotos twice?

Reply:
That's because I need two of every photo for the memory game! Perhaps there is a more elegant way of doing this but it works well!